### PR TITLE
Updates to source generators

### PR DESCRIPTION
**Pull request by andrewlock**
## Summary of changes

- Move `NativeObjects` generator to separate project
- Move all source generators into the root namespace
- Persist the source generator output for `dd_dotnet`

## Reason for change

- By referencing `dd_dotnet`, we source generate a bunch of stuff we don't need. That forces us to include dummy values etc, generally does more work than necessary, and all needs to be trimmed anyway.
- Our paths are long because the compiler includes the namespace in the output paths. To work around it, move the generators into the root namespace. We don't directly reference the generators anyway, so it doesn't change anything
- Persist the generator output in `dd_dotnet` (like we do for Datadog.Trace) so that we can track changes etc.

Also, our build is broken in GitLab and I can't merge https://github.com/DataDog/dd-trace-dotnet/pull/4642 until we fix this 😅 

## Implementation details

Pretty self explanatory

## Test coverage

All covered by existing tests. This is all just refactoring

## Other details
I will do a subsequent PR to add our standard copyright notice to the generated files. I don't want to do that until I merge https://github.com/DataDog/dd-trace-dotnet/pull/4642  though, otherwise I'm in merge hell

### CODE REVIEW

File: tracer/src/Datadog.Trace.Tools.dd_dotnet.SourceGenerators/Datadog.Trace.Tools.dd_dotnet.SourceGenerators.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <IsRoslynComponent>true</IsRoslynComponent>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <!--  Using the root namespace so generators are all in the root + we can control how they're written to disk  -->
+    <RootNamespace></RootNamespace> 
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="System.Reflection.Emit" Version="4.3.0" />
+    <PackageReference Include="System.Reflection.Emit.Lightweight" Version="4.3.0" />
```
Comment (kevingosse) [inline tracer/src/Datadog.Trace.Tools.dd_dotnet.SourceGenerators/Datadog.Trace.Tools.dd_dotnet.SourceGenerators.csproj:12]: Do we need those? 🤔

**Review by kevingosse — APPROVED** [review]

File: tracer/src/Datadog.Trace.Tools.dd_dotnet.SourceGenerators/Datadog.Trace.Tools.dd_dotnet.SourceGenerators.csproj (hunk 1 above)
Comment (andrewlock) [reply tracer/src/Datadog.Trace.Tools.dd_dotnet.SourceGenerators/Datadog.Trace.Tools.dd_dotnet.SourceGenerators.csproj:12]: Oops, no, probably not. That's probably related to the messagepack stuff

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (4660)](https://github.com/DataDog/dd-trace-dotnet/tree/d21eefc08b12ab7a072bd33b8cecac428001efb7)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/2c8d766f0aff6122774e8eed1c4457f7a4789f1a)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (69ms)  : 65, 73
     .   : milestone, 69,
    master - mean (68ms)  : 65, 72
     .   : milestone, 68,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (1,041ms)  : 1011, 1071
     .   : milestone, 1041,
    master - mean (1,034ms)  : 1009, 1059
     .   : milestone, 1034,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (106ms)  : 100, 113
     .   : milestone, 106,
    master - mean (105ms)  : 101, 110
     .   : milestone, 105,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (753ms)  : 730, 775
     .   : milestone, 753,
    master - mean (746ms)  : 725, 767
     .   : milestone, 746,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (90ms)  : 84, 97
     .   : milestone, 90,
    master - mean (90ms)  : 83, 97
     .   : milestone, 90,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (706ms)  : 684, 729
     .   : milestone, 706,
    master - mean (705ms)  : 684, 726
     .   : milestone, 705,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (190ms)  : 186, 194
     .   : milestone, 190,
    master - mean (191ms)  : 187, 195
     .   : milestone, 191,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (1,140ms)  : 1117, 1163
     .   : milestone, 1140,
    master - mean (1,147ms)  : 1127, 1167
     .   : milestone, 1147,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (276ms)  : 269, 284
     .   : milestone, 276,
    master - mean (278ms)  : 272, 283
     .   : milestone, 278,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (1,112ms)  : 1076, 1148
     .   : milestone, 1112,
    master - mean (1,110ms)  : 1072, 1148
     .   : milestone, 1110,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (4660) - mean (264ms)  : 260, 269
     .   : milestone, 264,
    master - mean (265ms)  : 258, 271
     .   : milestone, 265,

    section CallTarget+Inlining+NGEN
    This PR (4660) - mean (1,056ms)  : 1018, 1095
     .   : milestone, 1056,
    master - mean (1,054ms)  : 1023, 1086
     .   : milestone, 1054,

```

**Review by daniel-romano-DD — APPROVED** [review]
🫠

**Pull request merged**